packages/html-reporter/src/index.tsx

```diff
@@ -14,14 +14,50 @@
  * limitations under the License.
  */
 
-import type { HTMLReport } from '@playwright/test/src/reporters/html';
+import type { HTMLReport, TestAttachment } from '@playwright/test/src/reporters/html';
 import type zip from '@zip.js/zip.js';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import './colors.css';
 import { LoadedReport } from './loadedReport';
 import { ReportView } from './reportView';
 
+export type Metadata = Partial<{
+  'generatedAt': number;
+  'revision.id': string;
+  'revision.author': string;
+  'revision.email': string;
+  'revision.subject': string;
+  'revision.timestamp': number;
+  'revision.link': string;
+  'revision.localPendingChanges': boolean;
+  'ci.link': string;
+}>;
+
+const extractMetadata = (attachments: TestAttachment[]): Metadata | undefined => {
+  const field = (name: string) => attachments.find(({ name: n }) => n === name)?.body;
```

Comment:
nit: maybe `findLast` would be more natural vs `find` to prefer last plugin attachment?